fury/text_tools.py

```diff
@@ -0,0 +1,178 @@
+num_rows_ascii = 11
+
+
+def get_ascii_chars():
+    """This function is used to generate a list of ascii characters.
+
+    Returns
+    -------
+    chars : ndarray
+        A numpy array of ascii characters.
+    char2pos : dict
+        A dictionary that maps characters to their positions in the
+        numpy array.
+
+    """
+    ascii_chars = [
+        chr(i) for i in range(32, 127)
+    ]
+
+    chars = np.zeros(shape=(num_rows_ascii, num_cols_ascii), dtype='str')
+    char2pos = {}
+    for i in range(num_rows_ascii):
+        for j in range(num_cols_ascii):
+            index = i*num_cols_ascii+j
+            if index >= num_ascii_chars-1:
+                chars[i, j] = ' '
+                continue
+            char = ascii_chars[index]
+            chars[i, j] = char
+            char2pos[char] = [i/num_rows_ascii, j/num_cols_ascii][::-1]
+    return [
+        chars,
+        char2pos,
+    ]
+
+
+def create_bitmap_font(
+        font_size=50, font_path=None, pad=0,
+        show=False, save_path=None):
+    """This function is used to create a bitmap font.
+
+    Parameters
+    ----------
+    font_size : int
+        The size of the font.
+    font_path : str
+        The path to the font file.
+    pad : int
+        The padding of the font.
+    show : bool
+        Whether to show the result.
+    save_path : str
+        The path to save the image file.
+
+    Returns
+    -------
+    image_array : ndarray
+        The image array.
+    char2pos : dict
+        A dictionary that maps characters to their positions in the
+        numpy array.
+
+    """
+
+    chars, char2pos = get_ascii_chars()
+    if font_size == 50 and font_path is None:
+        font_path = f'{fury.__path__[0]}/data/files/font.bmp'
+        image_arr = Image.open(font_path)
+    else:
+        if font_path is None:
+            font_path = f'{fury.__path__[0]}/data/files/FreeMono.ttf'
+
+        width = num_cols_ascii*(font_size + pad*2)
+        height = num_rows_ascii*(font_size + pad*2)
+        # image = Image.new("RGB", (width, height), 'black')
+        # image = Image.new("RGBA", (width, height), (255, 255, 255, 0))
+        image = Image.new("P", (width, height))
+        draw = ImageDraw.Draw(image)
+        font = ImageFont.truetype(font_path, font_size)
+        for i_row, row in enumerate(chars):
+            x = 2*pad
+            for i_col, char in enumerate(row):
+                # draw.text((x, font_size*i_row + 2*pad), char, font=font)
+                draw.text(
+                    (x, font_size*i_row + 2*pad),
+                    # char, fill=(255, 255, 255, 255), font=font)
+                    char, fill=(255), font=font)
+                x += font_size
+        if show:
+            image.show()
+        if save_path is not None:
+            image.save(save_path)
+
+        image_arr = np.array(image)
+    # due vtk
+    image_arr = np.flipud(image_arr)
+    return image_arr, char2pos
+
+
+def get_positions_labels_billboards(
```

Comment:
This function is where python has the worst performance in this PR.  Maybe in the future we must to create a optimized version using Numba LLVM compiler or cython.

Reply:
In my testes this is not that bad, texture atlases are going to be created if they are not available for the first time, we should indicate the user what is happening.

Reply:
At the first time this will not create a texture atlas. The FreeMono Atlas is already inside of this PR. This will create a new texture atlas only if the user requested a atlas with a better resolution or a different font.

Reply:
I've changed that. Now is not possible to create a new atlas direct from` billboard_labels` actor. Instead you are allowed only to pass the name of the font to be used, `font_name`. 

The user can see the fonts available calling `text_tools.list_available_fonts()`.  

To create a new atlas to be used in the `billboard_lablels` you must explicit call `create_atlas_font`  method

```python
font_path = '/home/my_user/.../my_font.ttf'
text_tools.create_atlas_font(
    'NameOfMyNewFont', font_path=font_path, font_size_res=7)
# this will create a directory /home/.fury/font_atlas/NameOfMyNewFont
```
After this,  `NameOfMyNewFont` will be available
```python
new_font_actor = actor.bitmap_labels(
      [centers[5]], ['font size and format'],
      font_name='NameOfMyNewFont',
      align='center', scales=.05, colors=colors[3])
 ```